code/modules/clothing/clothing.dm

```diff
@@ -1197,61 +1198,64 @@
 
 /obj/item/clothing/under/examine(mob/user, distance, is_adjacent)
 	. = ..()
-	switch(src.sensor_mode)
-		if(0)
-			to_chat(user, "Its sensors appear to be disabled.")
-		if(1)
-			to_chat(user, "Its binary life sensors appear to be enabled.")
-		if(2)
-			to_chat(user, "Its vital tracker appears to be enabled.")
-		if(3)
-			to_chat(user, "Its vital tracker and tracking beacon appear to be enabled.")
-
-/obj/item/clothing/under/proc/set_sensors(mob/usr as mob)
-	var/mob/M = usr
-	if(M.stat || M.paralysis || M.stunned || M.weakened || M.restrained())
-		to_chat(usr, "You cannot reach your suit sensors like this...")
-		return
-	if(has_sensor >= 2)
-		to_chat(usr, "The controls are locked.")
+	if(has_sensor)
+		switch(src.sensor_mode)
+			if(SUIT_SENSOR_OFF)
+				to_chat(user, "Its sensors appear to be disabled.")
+			if(SUIT_SENSOR_BINARY)
+				to_chat(user, "Its binary life sensors appear to be enabled.")
+			if(SUIT_SENSOR_VITAL)
+				to_chat(user, "Its vitals tracker appears to be enabled.")
+			if(SUIT_SENSOR_TRACKING)
+				to_chat(user, "Its vitals tracker and tracking beacon appear to be enabled.")
+
+/obj/item/clothing/under/proc/set_sensors(mob/user as mob)
+	var/mob/M = user
+	if (isobserver(M)) return
+	if (user.incapacitated()) return
+	if(has_sensor >= SUIT_LOCKED_SENSORS)
+		to_chat(user, "The controls are locked.")
 		return 0
-	if(has_sensor <= 0)
-		to_chat(usr, "This suit does not have any sensors.")
+	if(has_sensor <= SUIT_NO_SENSORS)
+		to_chat(user, "This suit does not have any sensors.")
 		return 0
 
-	var/list/modes = list("Off", "Binary sensors", "Vitals tracker", "Tracking beacon")
-	var/switchMode = tgui_input_list(usr, "Select a sensor mode.", "Suit Sensor Mode", modes)
-	if(get_dist(usr, src) > 1)
-		to_chat(usr, "You have moved too far away.")
+	var/switchMode = tgui_input_list(user, "Select a sensor mode.", "Suit Sensor Mode", get_key_by_index(SUIT_SENSOR_MODES, sensor_mode + 1))
+	if(get_dist(user, src) > 1)
+		to_chat(user, "You have moved too far away.")
 		return
-	sensor_mode = modes.Find(switchMode) - 1
+	sensor_mode = SUIT_SENSOR_MODES[switchMode]
```

Comment:
Does this handle cancellation correctly, eg. if the input is cancelled, does this abort aswell, runtimes, something else?

Reply:
Yes